xl-objects.go

```diff
@@ -314,16 +405,35 @@ func (xl xlObjects) DeleteObject(bucket, object string) error {
 	}
 	// Wait for all the deletes to finish.
 	wg.Wait()
-	// Loop through and validate if any errors, return back the first
-	// error occurred.
-	for index, err := range errs {
+	// Loop through and validate if any errors, if we are unable to remove any part return
+	// "unexpected" error as returning any other error might be misleading. For ex.
+	// if DeleteFile() had returned errFileNotFound and we return it, then client would see
+	// ObjectNotFound which is misleading.
+	for _, err := range errs {
 		if err != nil {
-			partFileName := partNumToPartFileName(info.Parts[index].PartNumber)
-			return toObjectErr(err, bucket, pathJoin(object, partFileName))
+			return errUnexpected
 		}
 	}
 	err = xl.storage.DeleteFile(bucket, pathJoin(object, multipartMetaFile))
 	if err != nil {
+		return err
```

Comment:
this is the deleteObject function, not DeleteObject. DeleteObject() returns toObjectErr()